### PR TITLE
[REVIEW] Add more NVTX annotations in cuDF Python

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

**Comment by beckernick** [comment]
cc @jrhemstad

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -422,6 +423,7 @@ def __getattr__(self, key):
 
         raise AttributeError("'DataFrame' object has no attribute %r" % key)
 
+    @annotate("CUDF_DATAFRAME_GETITEM", color="blue", domain="cudf_python")
```
Comment (jrhemstad) [inline python/cudf/cudf/core/dataframe.py:426]: Like I said on the other PR, the `CUDF` prefix is really unnecessary when using the `cudf_python` domain.

**Comment by jrhemstad** [comment]
I have no notion of what is important to annotate in Python. If @beckernick and @shwina think these are the important things, then we can try a profile with these additional ranges and see what gaps are left.

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (shwina) [reply python/cudf/cudf/core/dataframe.py]: Renamed them all

File: python/cudf/cudf/io/csv.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from cudf.utils import ioutils
 
 
-@annotate("CUDF_READ_CSV", color="purple", domain="cudf_python")
+@annotate("READ_CSV", color="purple", domain="cudf_python")
```
Comment (kkraus14) [inline python/cudf/cudf/io/csv.py:10]: Can we add read / write parquet/orc as well?

Comment (kkraus14) [reply python/cudf/cudf/io/csv.py:10]: Can do in follow up

**Pull request merged**